src/celeritas/optical/ScintillationParams.cc

```diff
@@ -27,6 +29,16 @@ ScintillationParams::ScintillationParams(ScintillationInput const& input)
     for (auto const& spec : input.data)
     {
         CELER_EXPECT(spec.size() > 0);
+
+        // Validity of scintillation component data
+        real_type total_yield{0};
+        for (auto const& comp : spec)
+        {
+            CELER_EXPECT(comp);
+            total_yield += comp.yield_prob;
+        }
+        CELER_ENSURE(total_yield == 1);
```

Comment:
```suggestion
        CELER_ASSERT(spec.size() > 0);

        // Validity of scintillation component data
        real_type total_yield{0};
        for (auto const& comp : spec)
        {
            CELER_ASSERT(comp);
            total_yield += comp.yield_prob;
        }
        CELER_ASSERT(total_yield == 1);
```
We may need to do something like `CELER_ASSERT(soft_equal(real_type(1), total_yield))`, or renormalize the yields to make sure it's as close to 1 as possible.

Reply:
Thanks! Fixed.